### PR TITLE
QET-T15262

### PR DESCRIPTION
**Pull request by silvadf9**
Scenario for SF Adapter integration. Placing and receiving calls on SF Adapter.

### CODE REVIEW

File: page_objects/call_interaction_page.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def __init__(self, driver=None):
         self.caller_contact = "//span[@class='info-contact']"
         self.interaction_tab = "//li[contains(@id, 'context')]/a"
         self.history_tab = "//li[contains(@id, 'history')]/a"
-        self.script_tab = "//li[contains(@id, 'script')]/a"
+        self.script_tab = "//li[contains(@id, 'script')]/a | //button[@id='script_btn']"
```
Comment (iagolivino1) [inline page_objects/call_interaction_page.py:23]: you can use the `script_button` from `adapter_page.py`

File: page_objects/common_page.py [hunk 2]
```diff
@@ -6,10 +6,11 @@ def __init__(self, driver=None):
         self.driver = driver
         self.connector_url = None
         self.modal_dialog = "//div[@class='f9-modal-dialog'][contains(., '<title>')]"
-        self.modal_submit_button = "//button[contains(@class, 'f9-positive-cta-btn') and contains(., '<text>')]"
+        self.modal_submit_button = "//button[contains(@class, 'f9-positive-cta-btn') and contains(., '<text>')] | //button[contains(@id, 'connect_station_btn') and contains(., '<text>')] | //button[contains(@id, 'confirm-softphone-setup') and contains(., '<text>')] | //button[contains(@id, 'skills_confirm_btn') and contains(., '<text>')]"
```
Comment (iagolivino1) [inline page_objects/common_page.py:9]: does not the `confirm_selection_button` from `adapter_page.py` work?
maybe you can use the "When I confirm the...(skills, station)" steps. (let's discuss)

File: step_definitions/adapters/adapter_login_steps.py [hunk 3]
```diff
@@ -14,13 +14,22 @@
 }
 EXTENSION_NAME = "Five9 Agent Desktop Toolkit"
 
+def check_force_login(agent):
+    try:
+        common.wait_element_to_be_clickable(ADAPTER_LOGIN_PAGE.driver,ADAPTER_LOGIN_PAGE.force_login_button,5)
+        ADAPTER_LOGIN_PAGE.get_force_login_button().click()
+        return
+    except:
```
Comment (iagolivino1) [inline step_definitions/adapters/adapter_login_steps.py:22]: let's try to catch the exceptions individually. this way we can debug faster when some error happen.

File: step_definitions/adapters/adapter_steps.py [hunk 4]
```diff
@@ -20,6 +20,13 @@ def handle_adapter_dnc_number():
     except TimeoutException:
         print("DNC dialog did not appear!")
 
+def handle_call():
+    ADAPTER_PAGE.driver.refresh()
+    common.system_wait(5)
+    # common.wait_element_to_not_be_displayed(CALL_INTERACTIONS.driver, CALL_INTERACTIONS.softphone_iframe)
```
Comment (iagolivino1) [inline step_definitions/adapters/adapter_steps.py:26]: remove the commented line :)

File: step_definitions/call_interaction_steps.py [hunk 5]
```diff
@@ -30,6 +33,11 @@ def handle_dnc_dialog(action='accept', timeout=5, force=False):
     except TimeoutException:
         print("DNC dialog did not appear!")
```
Comment (iagolivino1) [inline step_definitions/call_interaction_steps.py:35]: add one more break line here please :)

File: page_objects/call_interaction_page.py (hunk 1 above)
Comment (silvadf9) [reply page_objects/call_interaction_page.py:23]: forgot to removed this change. Fixed.

File: page_objects/common_page.py (hunk 2 above)
Comment (silvadf9) [reply page_objects/common_page.py:9]: You're right, I should use use the "When I confirm the...(skills, station)" steps.
Fixed

File: step_definitions/adapters/adapter_steps.py (hunk 4 above)
Comment (silvadf9) [reply step_definitions/adapters/adapter_steps.py:26]: Fixed

File: step_definitions/call_interaction_steps.py (hunk 5 above)
Comment (silvadf9) [reply step_definitions/call_interaction_steps.py:35]: Fixed

File: step_definitions/adapters/adapter_login_steps.py (hunk 3 above)
Comment (silvadf9) [reply step_definitions/adapters/adapter_login_steps.py:22]: Fixed

**Review by iagolivino1 — APPROVED** [review]

**Pull request merged**